src/chatClient.cpp

```diff
@@ -3546,23 +3546,43 @@ void GroupChatRoom::enablePreview(uint64_t ph)
 
 bool GroupChatRoom::publicChat() const
 {
-    return (mChat->crypto()->isPublicChat());
+    if (mChat)
```

Comment:
Is it not guaranteed that this method is called only once the `mChat` is initialized?? Otherwise, there should be a bug, and this change (along with others in `getPublicHandle()`, `getNumPreviewers()` and `previewMode()` may have unexpected consequences, since the returned value will be affected by race conditions. @alber2510 , can you explain, please?

Reply:
We assume that the lambda using this method at the `Member` will not call it before `mChat` is initialized, but just in case we decided to avoid a potential race and avoid the segfault, but notify MEGA with an event.